### PR TITLE
Generalize the Semiconductor Interface

### PR DESCRIPTION
**Pull request by prkkumar**
Implementation based on [http://dx.doi.org/10.1063/1.4825209](http://dx.doi.org/10.1063/1.4825209).

Verification: MIS case from [https://www.nature.com/articles/s41598-020-66313-1](https://www.nature.com/articles/s41598-020-66313-1)

**FerroX:**
image
**Reference: (blue curve)**
image

### CODE REVIEW

**Comment by prkkumar** [comment]
Note that: This will change the benchmarks for the Regression Tests

File: Source/Solver/ChargeDensity.H [hunk 1]
```diff
@@ -12,3 +12,4 @@ void ComputeRho(MultiFab&      PoissonPhi,
                 MultiFab&      p_den,
                 const MultiFab& MaterialMask);
 
+
```
Comment (ajnonaka) [inline Source/Solver/ChargeDensity.H]:
```suggestion
```

**Review by ajnonaka — APPROVED** [review]
Looks good to me - approved!

File: Source/FerroX.cpp [hunk 2]
```diff
@@ -462,12 +466,26 @@ void InitializeFerroXNamespace(const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM
...
-     pp.query("intrinsic_carrier_concentration",intrinsic_carrier_concentration);
+     //The most common acceptor dopant in bulk Si is boron (B), which has Ea = 44 meV
+     //The most common donors in bulk Si are phosphorus (P) and arsenic (As), 
+     //which have ionization energies of Ed = 46 meV and 54 meV, respectively.
 
-     use_Fermi_Dirac = 0;
+     acceptor_ionization_energy = 44.0e-3; 
+     donor_ionization_energy = 46.0e-3; 
+
+     intrinsic_carrier_concentration = std::sqrt(Nc*Nv)*exp(-0.5*bandgap/(kb*T));
```
Comment (prkkumar) [inline Source/FerroX.cpp]:
```suggestion
     intrinsic_carrier_concentration = std::sqrt(Nc*Nv)*exp(-0.5*q*bandgap/(kb*T));
```

**Comment by prkkumar** [comment]
Thanks for approving, Andy! I will merge it now. Will start the drift-diffusion implementation off of fresh development branch.

**Pull request merged**